core/src/main/java/org/elasticsearch/index/fielddata/ScriptDocValues.java

```diff
@@ -147,9 +174,29 @@ public int size() {
             return values.count();
         }
 
+        /**
+         * Fetch the value at an index wrapped in a {@link ReadableDateTime}. We take care not to allocate a new {@link MutableDateTime}
+         * if possible.
+         */
+        private ReadableDateTime dateAt(int index, long value) {
+            if (dates == null) {
+                dates = new MutableDateTime[ArrayUtil.oversize(max(1, values.count()), RamUsageEstimator.NUM_BYTES_OBJECT_REF)];
+                dates[index] = new MutableDateTime(value, DateTimeZone.UTC);
+            } else {
+                if (index >= dates.length) {
```

Comment:
Can we just keep it simple here and eagerly allocate the array to full size (and fill it completely) above? These should be small sizes (single digits). I don't think we need to optimize as if this array could be hundreds of elements.